model-engine/model_engine_server/inference/vllm/vllm_server.py

```diff
@@ -200,23 +253,63 @@ def extract_logprobs(logprobs: Dict[int, Logprob]) -> Dict[int, float]:
     return [extract_logprobs(logprobs) for logprobs in output_logprobs]
 
 
+def parse_args():
+    parser = make_arg_parser()
+    return parser.parse_args()
+
+
 if __name__ == "__main__":
     check_unknown_startup_memory_usage()
+
     parser = argparse.ArgumentParser()
     parser.add_argument("--host", type=str, default=None)  # None == IPv4 / IPv6 dualstack
     parser.add_argument("--port", type=int, default=5005)
     parser = AsyncEngineArgs.add_cli_args(parser)
-    args = parser.parse_args()
+    args = parse_args()
+
+    logger.info("vLLM version %s", VLLM_VERSION)
+    logger.info("args: %s", args)
+
+    if args.served_model_name is not None:
+        served_model_names = args.served_model_name
+    else:
+        served_model_names = [args.model]
+
+    signal.signal(signal.SIGUSR1, debug)
 
     engine_args = AsyncEngineArgs.from_cli_args(args)
     engine = AsyncLLMEngine.from_engine_args(engine_args)
 
-    signal.signal(signal.SIGUSR1, debug)
+    event_loop: Optional[asyncio.AbstractEventLoop]
+    try:
+        event_loop = asyncio.get_running_loop()
```

Comment:
when would there be event loop?

Reply:
I guess for our use case, there wouldn't be one